### PR TITLE
feat(testing): Add test projects for Cloudflare Workers. Vercel Remix + NextJS, Deno, and Bun

### PR DESCRIPTION
**Pull request by max-stytch**
Adds a new `test-runtimes` directory with setup instructions for various runtimes the Node SDK can be used in. Hopefully this will make it easier to test iterations of the Node SDK in different runtimes. Runtimes are _not_ plug&play, developers will need to install various things, make accounts with providers, etc.

The following runtimes are tested:
- `cloudflare-workers`
- `remix`
- `nextjs serverless`
- `nextjs edge`
- `deno`
- `bun`

### CODE REVIEW

File: test-runtimes/nextjs/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+First, run the development server:
+
+```bash
+npm run dev
+# or
+yarn dev
+# or
+pnpm dev
+# or
+bun dev
```
Comment (chris-stytch) [inline test-runtimes/nextjs/README.md:14]: Typo?

File: test-runtimes/nextjs/src/pages/api/edge.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { NextResponse } from "next/server";
+
+import * as stytch from "stytch";
+
+const client = new stytch.Client({
+  // Find these values at https://stytch.com/dashboard/api-keys
+  // These ones are MADE UP!
+  project_id: "project-live-c60c0abe-c25a-4472-a9ed-320c6667d317",
+  secret: "secret-live-80JASucyk7z_G8Z-7dVwZVGXL5NT_qGAQ2I=",
```
Comment (chris-stytch) [inline test-runtimes/nextjs/src/pages/api/edge.ts:9]:
```suggestion
  project_id: "YOUR_STYTCH_PROJECT_ID",
  secret: "YOUR_STYTCH_SECRET",
```

This is the pattern we use elsewhere, dealer's choice on which you'd prefer.

File: test-runtimes/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Exclude lockfiles for packaging tests to reduce file churn
+/*/package-lock.json
```
Comment (chris-stytch) [inline test-runtimes/.gitignore:2]: Is there a preference for this over `.nvmrc` for each project and checking in `package-lock.json`?

File: test-runtimes/bun/index.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    project_id: "project-live-c60c0abe-c25a-4472-a9ed-320c6667d317",
+    secret: "secret-live-80JASucyk7z_G8Z-7dVwZVGXL5NT_qGAQ2I=",
+  }).magicLinks
+    .authenticate({
+      session_token: "WJtR5BCy38Szd5AfoDpf0iqFKEt4EE5JhjlWUY7l3FtY",
+    })
+    .catch((err: unknown) => console.log(JSON.stringify(err)));
+}
+
+doStytchRequest();
```
Comment (jeremy-stytch) [inline test-runtimes/bun/index.ts:14]: Should this this Promise be `await`-ed or `.then()`'d? Does Bun do some magic?

File: test-runtimes/deno/main.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  }).magicLinks
+    .authenticate({
+      session_token: "WJtR5BCy38Szd5AfoDpf0iqFKEt4EE5JhjlWUY7l3FtY",
+    })
+    .catch((err) => console.log(JSON.stringify(err)));
+}
+
+// Learn more at https://deno.land/manual/examples/module_metadata#concepts
+if (import.meta.main) {
+  doStytchRequest();
```
Comment (jeremy-stytch) [inline test-runtimes/deno/main.ts:16]: Similar to Bun, should this have `await`/`.then()`? Or is there magic?

**Review by jeremy-stytch — APPROVED** [review]
It looks like we commit the lockfiles from each environment-specific package manager, but not any of the `package-lock.json` files. Is that an important distinction? 

(I've never used any of Bun, Deno, etc., but I wouldn't be surprised if the answer is "`package-lock.json` is derived from the other lockfile".)

File: test-runtimes/nextjs/README.md (hunk 1 above)
Comment (max-stytch) [reply test-runtimes/nextjs/README.md:14]: This is the NextJS OOTB readme - they support bun too

File: test-runtimes/nextjs/src/pages/api/edge.ts (hunk 2 above)
Comment (max-stytch) [reply test-runtimes/nextjs/src/pages/api/edge.ts:9]: The secrets I'm checking in here are a special set we used to have public on the homepage - they produce a specific error and make it easy to confirm network connectivity. I'll add better comments.

File: test-runtimes/.gitignore (hunk 3 above)
Comment (max-stytch) [reply test-runtimes/.gitignore:2]: `.nvcmrc` for each project would mean many versions of Node, which we don't need... yet

File: test-runtimes/bun/index.ts (hunk 4 above)
Comment (max-stytch) [reply test-runtimes/bun/index.ts:14]: Bun does some magic! Bun won't end the process until the microtask queue is empty. No top-level await nesc.

**Comment by max-stytch** [comment]
> (I've never used any of Bun, Deno, etc., but I wouldn't be surprised if the answer is "package-lock.json is derived from the other lockfile".)

you are correct!

**Pull request merged**